frontend/server/src/main/java/org/pytorch/serve/wlm/Model.java

```diff
@@ -365,4 +381,25 @@ public long getClientTimeoutInMills() {
     public void setClientTimeoutInMills(long clientTimeoutInMills) {
         this.clientTimeoutInMills = clientTimeoutInMills;
     }
+
+    public boolean isUseJobTicket() {
+        return useJobTicket;
+    }
+
+    public int incNumJobTickets() {
+        return this.numJobTickets.incrementAndGet();
+    }
+
+    public int decNumJobTickets() {
+        return this.numJobTickets.decrementAndGet();
+    }
+
+    public synchronized boolean hasJobTickets() {
+        if (this.numJobTickets.get() == 0) {
+            return false;
+        }
+
+        this.numJobTickets.decrementAndGet();
```

Comment:
Why are we decreasing here when just checking if `hasJobTickets`

Reply:
multi-threads can query func hasJobTickets. Decreasing in this func is used to guarantee the ticket is available for this request.

Reply:
Could there potentially be a race condition here with two threads calling `hasJobTickets` simultaneously which could cause `numJobTickets` to become negative since at the time of checking both threads may not have completed the update just yet?

Reply:
this is a synchronized func.

Reply:
I see, then a function name like `getJobTicket` or `takeJobTicket` would make more sense, as it's actually trying to reduce the ticket count.